infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -281,26 +291,75 @@ public void resetUpgradeFlag() {
+        try {
+            Table<TableInfo, Namespace, CommonTypes.Uuid> streamsInfoTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE,
+                    LOG_REPLICATION_STREAMS_INFO_TABLE,
+                    TableInfo.class, Namespace.class, CommonTypes.Uuid.class,
+                    TableOptions.builder().build());
+
+            addStreamsToInfoTableWithRetry(streamIdSet, streamsInfoTable);
+        } catch (Exception e) {
+            log.warn("Exception when opening LR stream info table ", e);
+        }
+    }
+
+    private void addStreamsToInfoTable(Set<UUID> streamIdSet,
+                                       Table<TableInfo, Namespace, CommonTypes.Uuid> streamsInfoTable)
+            throws RetryNeededException {
+        try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+            for (UUID id : streamIdSet) {
+                TableInfo tableInfo = TableInfo.newBuilder().setId(id.toString()).build();
+                Namespace namespace = Namespace.newBuilder().setName(EMPTY_STR).build();
+                txn.putRecord(streamsInfoTable, tableInfo, namespace, defaultMetadata);
+            }
+            txn.commit();
+        } catch (TransactionAbortedException tae) {
+            throw new RetryNeededException();
+        }
+    }
+
+    private void addStreamsToInfoTableWithRetry(Set<UUID> streamIdSet,
+                                                Table<TableInfo, Namespace, CommonTypes.Uuid> streamsInfoTable) {
+        try {
+            IRetry.build(IntervalRetry.class, () -> {
+                addStreamsToInfoTable(streamIdSet, streamsInfoTable);
+                log.debug("Successfully added streams {} to the info table.", streamIdSet);
+                return null;
+            }).run();
+        } catch (InterruptedException ie) {
+            log.error("Unrecoverable exception when attempting to add streams.", ie);
+            throw new UnrecoverableCorfuInterruptedError(ie);
         }
-        return streamingConfig;
+    }
+
+    public Set<TableInfo> readStreamsToReplicateFromRegistry(long ts) {
+
+        Set<TableInfo> tableInfoSet = new HashSet<>();
+        CorfuTable<CorfuStoreMetadata.TableName, CorfuRecord<CorfuStoreMetadata.TableDescriptors,
+                CorfuStoreMetadata.TableMetadata>>
+                registryTable = runtime.getTableRegistry().getRegistryTable();
+        Token snapshotToken = Token.of(0L, ts);
+        runtime.getObjectsView().TXBuild()
+                .type(TransactionType.SNAPSHOT)
+                .snapshot(snapshotToken)
+                .build()
+                .begin();
+        Set<CorfuStoreMetadata.TableName> tableNameSet = registryTable.keySet();
+
+        for (CorfuStoreMetadata.TableName tableName : tableNameSet) {
+            CorfuRecord<CorfuStoreMetadata.TableDescriptors,
+                                CorfuStoreMetadata.TableMetadata> tableRecord = registryTable.get(tableName);
```

Comment:
just curious why we need another get() here and not retrieve the CorfuRecord in L348 itself?